mapbox_vector_tile/encoder.py

```diff
@@ -56,13 +56,28 @@ def transform(shape, func):
     raise ValueError('Unknown geometry type, "%s"' % shape.type)
 
 
+def on_invalid_geometry_raise(shape):
+    raise Exception('Invalid geometry: %s' % shape.wkt)
+
+
+def on_invalid_geometry_ignore(shape):
+    return None
+
+
+def on_invalid_geometry_make_valid(shape):
+    if shape.type in ('Polygon', 'MultiPolygon'):
+        shape = shape.buffer(0)
```

Comment:
I added an assertion in 6a1e8e0. A geometry validity monad could be an exercise for the reader :stuck_out_tongue_winking_eye: 

Reading the [shapely docs on buffer(0)](http://toblerity.org/shapely/manual.html) (search for bowtie), it looks like it may generate a MultiPolygon to make the geometry valid. Could this affect the winding order?